conans/test/functional/generators/components/cmake_find_package_multi_test.py

```diff
@@ -804,3 +804,146 @@ def build(self):
                       "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:>;"
                       "$<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:>",
                       client.out)
+
+    def test_same_name_global_target_collision(self):
+        conanfile_tpl = textwrap.dedent("""
+            import os
+            from conans import ConanFile, CMake
+
+            class Conan(ConanFile):
+                name = "{name}"
+                version = "1.0"
+                settings = "os", "compiler", "build_type", "arch"
+                generators = "cmake"
+                exports_sources = "src/*"
+
+                def build(self):
+                    cmake = CMake(self)
+                    cmake.configure(source_folder="src")
+                    cmake.build()
+
+                def package(self):
+                    self.copy("*.h", dst="include", src="src")
+                    self.copy("*.lib", dst="lib", keep_path=False)
+                    self.copy("*.a", dst="lib", keep_path=False)
+
+                def package_info(self):
+                    self.cpp_info.names["cmake_find_package_multi"] = "nonstd"
+                    self.cpp_info.filenames["cmake_find_package_multi"] = "{name}"
+                    self.cpp_info.components["1"].names["cmake_find_package_multi"] = "{name}"
+                    self.cpp_info.components["1"].libs = ["{name}"]
+            """)
+        client = TestClient()
+        for name in ["expected", "variant"]:
+            client.run("new {name}/1.0 -s".format(name=name))
+            client.save({"conanfile.py": conanfile_tpl.format(name=name)})
+            client.run("create . {name}/1.0@".format(name=name))
+        middle_cmakelists = textwrap.dedent("""
+            project(middle)
+            cmake_minimum_required(VERSION 3.1)
+
+            include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+            conan_basic_setup()
+
+            find_package(expected)
+            find_package(variant)
+
+            add_library(middle middle.cpp)
+            target_link_libraries(middle nonstd::expected nonstd::variant)
```

Comment:
I think here we need to test how the behavior changes.  Before if using `nonstd::nonstd` it should fail to link, with the changes in this PR it will work (both mini-targets are available).

Reply:
Do you mean to split this into two PRs? one adding a test failing and another one with the fix and the change in the test?

Reply:
It means that a test with ``nonstd::nonstd`` seems good, to prove it works now.
Probably no need another full test, just adding a new target, using the same file, would be enough.

Reply:
That's it, we need to test that using `nonstd::nonstd`, it works (Conan aggregates targets coming from different packages). We already know it works if we use explicitly those two mini-targets.

Reply:
...but, it is true this is linking statically, so the consumer needs to link with both (expected and variant), so this test would fail without theis fix.

Reply:
anyway, I have changed the targets for the ``nonstd::nonstd`` to make it more evident. thanks!